src/impl/collection_change_builder.hpp

```diff
@@ -24,9 +24,69 @@
 #include <realm/util/optional.hpp>
 
 #include <unordered_map>
+#include <unordered_set>
+#include <vector>
 
 namespace realm {
 namespace _impl {
+
+class ObjectChangeSet {
```

Comment:
This is part of the public API for bindings so it shouldn't be in the _impl namespace. CollectionChangeBuilder wasn't public because it was only the mutation part and CollectionChangeSet was the public read-only interface.